controllers/webhook/ksvc_validator.go

```diff
@@ -0,0 +1,113 @@
+	log.Info("Validating Knative Service")
+
+	// If there is an explicit intent for not having a sidecar, skip validation
+	ksvcTemplateMeta := ksvc.Spec.Template.GetObjectMeta()
+	if ksvcTemplateMeta != nil {
+		if templateAnnotations := ksvcTemplateMeta.GetAnnotations(); templateAnnotations[constants.IstioSidecarInjectAnnotationName] == "false" {
+			log.V(1).Info("Skipping validation of Knative Service because there is an explicit intent to exclude it from the Service Mesh")
+			return nil
+		}
+	}
+
+	// Only validate the KSVC if it is owned by KServe controller
+	ksvcMetadata := ksvc.GetObjectMeta()
+	if ksvcMetadata == nil {
+		log.V(1).Info("Skipping validation of Knative Service because it does not have metadata")
+		return nil
+	}
+	ksvcOwnerReferences := ksvcMetadata.GetOwnerReferences()
+	if ksvcOwnerReferences == nil {
+		log.V(1).Info("Skipping validation of Knative Service because it does not have owner references")
+		return nil
+	}
+	isOwnedByKServe := false
+	for _, owner := range ksvcOwnerReferences {
+		if owner.Kind == constants.InferenceServiceKind {
+			if strings.Contains(owner.APIVersion, kservev1beta1.SchemeGroupVersion.Group) {
+				isOwnedByKServe = true
+			}
+		}
+	}
+	if !isOwnedByKServe {
+		log.V(1).Info("Skipping validation of Knative Service because it is not owned by KServe")
+		return nil
+	}
+
+	// Since the Ksvc is owned by an InferenceService, it is known that it is required to be
+	// in the Mesh. Thus, the involved namespace needs to be enrolled in the mesh.
+	// Go and check the ServiceMeshMemberRoll to verify that the namespace is already a
+	// member. If it is still not a member, reject creation of the Ksvc to prevent
+	// creation of a Pod that would not be in the Mesh.
+	smmrQuerier := resources.NewServiceMeshMemberRole(v.client)
+	smmr, fetchSmmrErr := smmrQuerier.FetchSMMR(ctx, log, types2.NamespacedName{Name: constants.ServiceMeshMemberRollName, Namespace: constants.IstioNamespace})
+	if fetchSmmrErr != nil {
+		log.Error(fetchSmmrErr, "Error when fetching ServiceMeshMemberRoll", "smmr.namespace", constants.IstioNamespace, "smmr.name", constants.ServiceMeshMemberRollName)
+		return fetchSmmrErr
+	}
+	if smmr == nil {
+		log.Info("Rejecting Knative service because the ServiceMeshMemberRoll does not exist", "smmr.namespace", constants.IstioNamespace, "smmr.name", constants.ServiceMeshMemberRollName)
+		return fmt.Errorf("rejecting creation of Knative service %s on namespace %s because the ServiceMeshMemberRoll does not exist", ksvc.Name, ksvc.Namespace)
+	}
+
+	log = log.WithValues("smmr.namespace", smmr.Namespace, "smmr.name", smmr.Name)
+
+	if smmr.Status.Members == nil {
+		log.Info("Rejecting Knative service because the service mesh does not have any namespaces")
+		return fmt.Errorf("rejecting creation of Knative service %s on namespace %s because the namespace is yet not a member of the service mesh", ksvc.Name, ksvc.Namespace)
+	}
+	for _, memberNamespace := range smmr.Status.Members {
```

Comment:
I am not sure if it won't be better to check `ConfiguredMembers` and `PendingMembers` instead. I think `Members` is what is expected to be enrolled - so all known members disregards their statuses.

I also wonder if you don't want to check the actual state of things in `MemberStatuses` slice to get the full picture.

Reply:
I think this is good feedback.
I changed it to use `ConfigredMembers` since that is what is documented here: https://docs.openshift.com/rosa/service_mesh/v2x/ossm-create-mesh.html